src/pages/home/sidebar/OptionRowTitle/index.js

```diff
@@ -0,0 +1,120 @@
+import _ from 'underscore';
+import React, {Fragment, PureComponent} from 'react';
+import {
+    Text,
+    View,
+} from 'react-native';
+import {propTypes, defaultProps} from './OptionRowTitleProps';
+import styles from '../../../../styles/styles';
+import Tooltip from '../../../../components/Tooltip';
+import hasEllipsis from '../../../../libs/hasEllipsis';
+
+class OptionRowTitle extends PureComponent {
+    constructor(props) {
+        super(props);
+        this.ref = null;
+        this.cRefs = [];
+        this.setContainerRef = (ref) => {
+            this.ref = ref;
+        };
+        this.setOptionChildRef = index => (ref) => {
+            this.cRefs[index] = ref;
+        };
+        this.state = {
+            isEllipsisActive: false,
+            containerLayout: null,
+        };
+        this.setContainerLayout = this.setContainerLayout.bind(this);
+        this.getTooltipShiftX = this.getTooltipShiftX.bind(this);
+    }
+
+    componentDidMount() {
+        this.setState({
+            isEllipsisActive: this.ref && hasEllipsis(this.ref),
+        });
+    }
+
+    setContainerLayout({nativeEvent}) {
+        this.setState({
+            containerLayout: nativeEvent.layout,
```

Comment:
Rather than setting `containerLayout` which causes more verbose code elsewhere, it would be better to set `containerWidth` and `containerLeft`

Reply:
Does this need to be stored in the state at all, or can it be stored directly as class properties (eg. `this.containerWidth`)?